File: Source Code/Bike_Car_Rental_Portal/src/main/java/com/example/demo/entities/Owner.java
package com.example.demo.entities;

import jakarta.persistence.*;

@Entity
@Table(name = "owner")
public class Owner {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "owner_id")
    private int ownerId;

    @Column(name = "name", nullable = false, length = 100)
    private String name;

    @Column(name = "email", nullable = false, length = 100)
    private String email;

    @Column(name = "phone_number", nullable = false, length = 100)
    private String phoneNumber;

    @Column(name = "address", nullable = false, length = 100)
    private String address;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private Login user;  // Ensure this is correct

    // Getters and setters
    public int getOwnerId() { return ownerId; }
    public void setOwnerId(int ownerId) { this.ownerId = ownerId; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public String getPhoneNumber() { return phoneNumber; }
    public void setPhoneNumber(String phoneNumber) { this.phoneNumber = phoneNumber; }

    public String getAddress() { return address; }
    public void setAddress(String address) { this.address = address; }

    public Login getUser() { return user; }
    public void setUser(Login user) { this.user = user; }

    // Constructors
    public Owner() {}

    public Owner(String name, String email, String phoneNumber, String address, Login user) {
        this.name = name;
        this.email = email;
        this.phoneNumber = phoneNumber;
        this.address = address;
        this.user = user;
    }
}

File: Source Code/rentalSystem/rentalSystem/Models/Owner.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace rentalSystem.Models
{
    public partial class Owner
    {
        public Owner()
        {
            Vehicles = new HashSet<Vehicle>();
        }

        public int OwnerId { get; set; }
        public string Name { get; set; } = null!;
        public string Email { get; set; } = null!;

        public string PhoneNumber { get; set; }
        public string Address { get; set; } = null!;
        public int? UserId { get; set; }

        public virtual Login? User { get; set; }
        public virtual ICollection<Vehicle> Vehicles { get; set; }
    }
}
